jsonschema/validators.py

```diff
@@ -277,26 +308,20 @@ def resolving(self, ref):
         :argument str ref: reference to resolve
 
         """
+        url = self.urljoin_cache(self.resolution_scope, ref)
+        return url, self.resolve_cache(url)
 
-        full_uri = urljoin(self.resolution_scope, ref)
-        uri, fragment = urldefrag(full_uri)
-        if not uri:
-            uri = self.base_uri
-
-        if uri in self.store:
-            document = self.store[uri]
-        else:
+    def resolve_from_url(self, url):
+        ref = urldefrag(url)
+        try:
+            document = self.store[ref.url]
```

Comment:
Yup, it is intentional. Credit goes to ankostis for this one (this was from the cherry-pick).

While raised exceptions are slower, we actually expect the common case to be retrieving the document succesfully.  I believe in the success case it's actually faster to use `__getitem__` directly and try/except.